### PR TITLE
[ARO] az aro update: Add RBAC validation on cluster update

### PR DESCRIPTION
**Pull request by bennerv**
**Description**
Upstreams [Azure/ARO-RP#1159](https://github.com/Azure/ARO-RP/pull/1159) which adds RBAC validation on ARO cluster update, unblocking updates when customers remove SPN permissions.  

Upstreams [Azure/ARO-RP#1217](https://github.com/Azure/ARO-RP/pull/1159) which updates error handling in ARO to conform to the guidelines. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by bennerv** [comment]
cc @jim-minter  and @mjudeikis

File: src/azure-cli/azure/cli/command_modules/aro/_aad.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 logger = get_logger(__name__)
 
 
-class AADManager:
+class AADManager():
```
Comment (jim-minter) [inline src/azure-cli/azure/cli/command_modules/aro/_aad.py:22]: @bennerv I'd expect this to be `class AADManager(object):`

Comment (bennerv) [reply src/azure-cli/azure/cli/command_modules/aro/_aad.py:22]: @jim-minter `azdev style aro` complains about it when it's there.  I can remove the `()` if we think that's best so it shows no change.

```
************* Module azure.cli.command_modules.aro._aad
_aad.py:22:0: R0205: Class 'AADManager' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
```

**Comment by yonzhan** [comment]
ARO

**Comment by jim-minter** [comment]
@yonzhan lgtm, please merge

**Comment by yonzhan** [comment]
@fengzhou-msft will help.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**